qcodes/instrument_drivers/new_style.py

```diff
@@ -0,0 +1,161 @@
+"""Driver examples using the new @add_parameter decorator style.
```

Comment:
i really like the "conventional style"/"method style" and "decorator style" terminology - shall it be used everywhere in the docs, docstring, and module names? the example notebook is already using it, so i'd suggest to rename this module to, say, ``decorator_style``, and also skim thorugh the docstrings/code to replace "new/old-style" with "method/decorator-style" terms

Reply:
Agreed, "method style" and "decorator style"